src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.Helpers.cs

```diff
@@ -14,15 +16,32 @@ public static partial class JsonSerializer
             ref Utf8JsonReader reader)
         {
             ReadStack state = default;
-            if (options.ReferenceHandling.ShouldReadPreservedReferences())
-            {
-                state.ReferenceResolver = new DefaultReferenceResolver(writing: false);
-            }
-            state.Current.Initialize(returnType, options);
+            state.InitializeRoot(returnType, options);
 
             ReadCore(options, ref reader, ref state);
 
             return state.Current.ReturnValue;
         }
     }
+
+    internal static class JsonReaderExtensions
```

Comment:
nit: Consider
1) If we are keeping this, move to a separate file. OR
2) [preferred] Moving the methods to existing helper classes that are used in different layers of the stack like `JsonHelpers.cs`. I can see these methods be useful elsewhere (including in JsonDocument/etc.). Extension methods don't have to belong to a {Name}Extensions class.

Reply:
Sure. Moving to JsonHelpers.cs. I wonder if adding a public ReadWithVerify() would have value?